src/core/models/outputs/output_foundry.h

```diff
@@ -6,22 +6,104 @@
+} token_scheme_e;
 
 /**
  * @brief An output type which controls the supply of user defined native tokens.
  *
  */
 typedef struct {
-  void* address;                  ///< The alias controlling this foundry
-  uint64_t amount;                ///< The amount of IOTA tokens held by this output
-  native_tokens_t native_tokens;  ///< The native tokens held by this output
-  uint32_t serial;                ///< The serial number of the foundry
-  void* token_tag;                ///< The tag which is always the last 12 bytes of the tokens generated by this foundry
-  void* circulating_supply;       ///< The circulating supply of tokens controlled by this foundry
-  void* max_supply;               ///< The maximum supply of tokens controlled by this foundry
-  void* token_scheme;             ///< The token scheme used by this foundry
-  void* feature_blocks;           ///< The feature flocks which modulate the constrants on this output
+  address_t* address;                     ///< The alias controlling this foundry
+  uint64_t amount;                        ///< The amount of IOTA tokens held by this output
+  native_tokens_t* native_tokens;         ///< The native tokens held by this output
+  uint32_t serial;                        ///< The serial number of the foundry
+  byte_t token_tag[TOKEN_TAG_BYTES_LEN];  ///< The tag which is always the last 12 bytes of the tokens generated by this
+                                          ///< foundry
+  uint256_t circ_supply;                  ///< The circulating supply of tokens controlled by this foundry
+  uint256_t max_supply;                   ///< The maximum supply of tokens controlled by this foundry
+  token_scheme_e token_scheme;            ///< The token scheme used by this foundry
+  feat_blk_list_t* feature_blocks;        ///< The feature blocks which modulate the constrants on this output
 } output_foundry_t;
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Create new Foundry Output object.
+ *
+ * @param[in] addr The alias controlling the foundry
+ * @param[in] amount The amount of IOTA tokens held by the output
+ * @param[in] tokens The list of native tokens held by the output
+ * @param[in] serial The serial number of the foundry
+ * @param[in] token_tag The last 12 bytes of ID of the tokens
+ * @param[in] circ_supply The circulating supply of tokens controlled by the foundry
+ * @param[in] max_supply The maximum supply of tokens controlled by the foundry
+ * @param[in] token_scheme The token scheme of the foundry
+ * @param[in] feat_blocks Set of feature blocks
+ *
+ * @return output_foundry_t* or NULL on failure
+ */
+output_foundry_t* output_foundry_new(address_t* addr, uint64_t amount, native_tokens_t* tokens, uint32_t serial_num,
+                                     byte_t token_tag[], uint256_t circ_supply, uint256_t max_supply,
+                                     token_scheme_e token_scheme, feat_blk_list_t* feat_blocks);
```

Comment:
Should `circ_supply` and `max_supply` be pointers?

Reply:
Done.